variant-remapping-automation/remapping_process.nf

```diff
@@ -0,0 +1,137 @@
+    if (!params.species_name) log.warn('Provide a species name using --species_name')
+    if (!params.genome_assembly_dir) log.warn('Provide a path to where the assembly should be downloaded using --genome_assembly_dir')
+    exit 1, helpMessage()
+}
+
+
+
+species_name = params.species_name.toLowerCase().replace(" ", "_")
+
+
+process retrieve_source_genome {
+
+    output:
+    path "${params.source_assembly_accession}.fa" into source_fasta
+    path "${params.source_assembly_accession}_assembly_report.txt" into source_report
+
+    """
+    $params.executable.genome_downloader --assembly-accession ${params.source_assembly_accession} --species ${species_name} --output-directory ${params.genome_assembly_dir}
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.source_assembly_accession}/${params.source_assembly_accession}.fa
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.source_assembly_accession}/${params.source_assembly_accession}_assembly_report.txt
+    """
+}
+
+
+process retrieve_target_genome {
+
+    output:
+    path "${params.target_assembly_accession}.fa" into target_fasta
+    path "${params.target_assembly_accession}_assembly_report.txt" into target_report
+
+    """
+    $params.executable.genome_downloader --assembly-accession ${params.target_assembly_accession} --species ${species_name} --output-directory ${params.genome_assembly_dir}
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.target_assembly_accession}/${params.target_assembly_accession}.fa
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.target_assembly_accession}/${params.target_assembly_accession}_assembly_report.txt
+    """
+}
+
+
+/*
+ * Extract the submitted variants to remap from the accesioning warehouse and store them in a VCF file.
+ */
+process extract_vcf_from_mongo {
+    input:
+    path source_fasta from source_fasta
+    path source_report from source_report
+
+    output:
+    path "${params.source_assembly_accession}_extraction.properties" into extraction_props
+    // Store both vcfs (eva and dbsnp) into one channel
+    path "*.vcf" into source_vcfs
+    val "${params.source_assembly_accession}_vcf_extractor.log" into log_filename
+
+    """
+    cp ${params.template_properties} ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.fasta=${source_fasta}" >> ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.assemblyReportUrl=file://${source_report}"" >> ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.assemblyAccession=${params.source_assembly_accession}"" >> ${params.source_assembly_accession}_extraction.properties
+    echo "parameters.outputFolder=." >> ${params.source_assembly_accession}_extraction.properties
+
+    java -jar $params.jar.vcf_extractor --spring.config.name=${params.source_assembly_accession}_extraction.properties > ${params.source_assembly_accession}_vcf_extractor.log
```

Comment:
I don't know if this will matter in this case, but I found the logs redirection kind of painful while testing submission automation and opted instead to use [`clusterOptions`](https://github.com/EBIvariation/eva-submission/blob/master/nextflow/accession.nf#L123)...

Reply:
I think I still prefer the redirection as I'll get the same results on local and on the cluster. Happy to discuss pros and cons though.